### DISCUSSION TITLE
watchdog restart

### DISCUSSION THREAD

**Discussion by luciangpl** [arendst/Tasmota#14802] [Q&A] [answered]
Hi, is there a way to automatically force a device to restart it is losing the Wi-Fi connection?
I have an esp32 and from time to time (weeks), the webpage is not available, no ping response. If I restart it, everything goes back to normal.

**Comment by sfromis** [comment, marked as answer]
If you don't want to debug your issue, you can use a rule with a trigger like `Wifi#Disconnected` to react to wifi going from connected to disconnected.

**Reply by luciangpl** [reply]
Can you guide me a little? I presume I should send a command in console, can you tell me what that command should be for restarting the device if the wifi disconnects for 1 min or something like that?

**Reply by sfromis** [reply]
Rules means creating a rule, see https://tasmota.github.io/docs/Rules/

**Reply by ascillato** [reply]
**Recomendation:** Check your Wi-Fi network. Your devices shouldn't loose Wi-Fi if their signal is above 60%. In the docs and in old issues are bunch of information, videos and recommendations about that. This has been addressed several times before. Please, do a quick search in old issues.

But, if instead you want to do that with a rule _**(not recommended)**_ the rule is:

Go to the console and type:

```JS
Rule 1
Rule ON Wifi#Disconnected DO Restart 1 ENDON
```

**Reply by luciangpl** [reply]
Thank @ascillato, just tested it and it works !
